### ISSUE TITLE
Package name clash

### ISSUE THREAD

**Issue by alesuiss** [grindsa/acme2certifier#63]
Hello,

The "acme" package name clashes with the python-acme client library. This is especially annoying for me since I need to use another ACME responder as the back-end.

If you used relative imports in your code, I think I could work around the problem, but as it is, I'll have to rename the package in every file.

Thanks for your consideration.

**Comment by grindsa** [comment]
Hi,

let me think about it. Your request makes definitely sense but it seems to be a bigger change.  I would have a few questions:

- Seems you are using a ca_handler using [acme-python](https://acme-python.readthedocs.io/en/latest/). Is this understanding correct? If so, would you mind sharing an example allowing me to replicate the issue?
- Can you describe your deployment a bid more in detail. Containerized (yes/no), apache2/nginx, wsgi/django? 
- Do I understand correctly that your current workaround is to rename the `acme2certifier/acme` directory and modify all the files accordingly? If so, any chance to get a diff from any a2c version comparing your version with the original ones? This would help a lot to get a better understanding of the complexity of the change.

Thx and best regards
Joern

**Comment by alesuiss** [comment]
Hello,

- The idea is indeed to write a ca_handler that uses that library. The problem is that the first thing I have to write is something like `from acme import client`, which doesn't work as `acme` resolves to your package. But I'm really no expert in Python namespacing, so maybe there's a work-around I'm not thinking of.
- I'm at the POC stage still and not sure what the actual deployment strategy will be, but if I can be bothered to do it properly, it will be containerized on k8s (Openshift) with Django. ATM it's locally on my workstation and on test VMs, with no application server.
- The workaround (that I hadn't implemented yet) would be to have my own fork or your project, and rename the `acme` package to something like `acme2certifier`, which would be a huge pain for updates. Again, not super knowledgeable on Python scoping, so maybe there's a solution I don't know about.

Thanks! And btw thanks for the software, if things work out it will be a huge help for us :-)

**Comment by alesuiss** [comment]
Hello again,

FYI, I got my proof-of-concept working.

I've replaced imports in the acme2certifier sources with relative ones, e.g. `from acme.message import Message` -> `from .message import Message`. Then I can just point the PYTHONPATH one directory above, and your package becomes `acme2certifier.acme`; then everything works, and I can `import acme.client` in my CAhandler. I've implemented the `enroll` method so far, works like a charm!

So, if you wouldn't mind switching to relative imports, that would already be a big help for me, as I wouldn't need to track any changes and could use your package directly (with some deployment fiddling). But if you would consider renaming the package, I would gladly submit a PR of my ACME CAhandler in return once it's fully implemented :-) Although its functionality will be somewhat limited -- it doesn't answer ACME challenges (indeed my use case includes adding challenges to a commercial ACME endpoint that doesn't issue them).

Best,

**Comment by grindsa** [comment]
I prefer to rename the package as this sounds to be the better way forward. As said before, I expect this to be a complex thing and I would like to create a new release for this change only. My current plan is to finish the work on version 0.16 this week (i still have some smaller outstanding issue to fix) and start next with the renaming from next week onwards. I will create a separate branch to collect the different changes and to give you some code to play with.

/G

**Comment by grindsa** [comment]
started to work on it in [acme_srv branch](https://github.com/grindsa/acme2certifier/tree/acme_srv)

**Comment by grindsa** [comment]
all changes made into [version 0.17](https://github.com/grindsa/acme2certifier/releases/tag/0.17). Feel free to give it a try...

**Comment by grindsa** [comment]
Again thank you for your contributing to acme2certifier. I am currently working to extend your handler with account registration and (http) challenge validation capabilities to interoperate with  CAs like Letsencrypt and ZeroSSLs. The first results are looking promising require some smaller adjustments which will break interoperability with your implementation. 

Would you mind to tell me which commercial CA product you are using?

Further, the [new handler](https://github.com/grindsa/acme2certifier/blob/acme_ca_handler/examples/ca_handler/acme_ca_handler.py) should work in your environment with the below additional parameters in `acme_srv.cfg`:

```cfg
account_path: /account/
directory_path: /
````

Could you give it a try? I created a [dedicated branch](https://github.com/grindsa/acme2certifier/tree/acme_ca_handler) however it should be enough to replace the handler with the one I linked above.

**Comment by alesuiss** [comment]
Hello,

Cool! I'll try to look at it next week. The service we use in the back-end is Sectigo.

How do you plan to solve http challenge validation though? With an endpoint like Let's Encrypt it will want to issue the challenge to the DNS name in the certificate request, i.e. presumably the calling client; I don't see how that would work.

Cheers,

**Comment by grindsa** [comment]
Acme2certifer acts as endpoint for challenge validation. Thus: it needs to be accessible by Letsencrypt. (I configured some restrictive port forwarding on our edge-routers)
I solved the challenge validation problem by separating internal and external DNS name spaces and by adding a wild-card entry on my external DNS server pointing to the acme2certifier system.  

I added a bid of  [documentation](https://github.com/grindsa/acme2certifier/blob/acme_ca_handler/docs/acme_ca.md) this morning and hope its not too clumsy.
